File: src/exceptions.rs
use pyo3::create_exception;

// Base Exception for all songbird errors
create_exception!(module, SongbirdError, pyo3::exceptions::PyException);

create_exception!(module, UseAsyncConstructorError, SongbirdError);
create_exception!(module, CouldNotConnectToRTPError, SongbirdError);
create_exception!(module, ConsumedSourceError, SongbirdError);
create_exception!(module, CouldNotOpenFileError, SongbirdError);
create_exception!(module, YtdlError, SongbirdError);
create_exception!(module, FfmpegError, SongbirdError);
create_exception!(module, TrackError, SongbirdError);

File: songbird/exceptions.py
from .songbird import (
    YtdlError,
    FfmpegError,
    SongbirdError,
    ConsumedSourceError,
    CouldNotOpenFileError,
    UseAsyncConstructorError,
)


class QueueError(SongbirdError):
    """Queue raises an exception"""


__all__ = [
    "CouldNotOpenFileError",
    "FfmpegError",
    "YtdlError",
    "ConsumedSourceError",
    "UseAsyncConstructorError",
    "QueueError",
]

